lib/modules/apostrophe-templates/index.js

```diff
@@ -696,5 +696,29 @@ module.exports = {
       req.data.aposBodyClasses = (req.data.aposBodyClasses ? (req.data.aposBodyClasses + ' ') : '') +
         bodyClass;
     };
+
+    // Add a body attribute to be emitted when the page is rendered. This information
+    // is attached to `req.data`, where `req.data.aposBodyDataAttributes` is built up
+    // using `name` as the attribute name which is automatically prepended with "data-"
+    // and the optional `value` argument
+    // Also receives a single object or an array of objects with name and value key pairs
+    // The default `outerLayoutBase.html` template outputs that string.
+
+    self.addBodyDataAttribute = function(req, name, value = '') {
+      var values;
+      if (_.isObject(name) && !_.isArray(name) && !_.isFunction(name)) {
+        values = [name];
+      } else {
+        values = !_.isArray(name) ? [{name: name, value: value}] : name;
+      }
+      _.each(values, function(attribute) {
+        attribute.name = _.trim(attribute.name);
+        if (_.isEmpty(attribute.name)) {
+          return;
+        }
+        req.data.aposBodyDataAttributes = (req.data.aposBodyDataAttributes ? (req.data.aposBodyDataAttributes + ' ') : ' ') +
+        'data-' + (!_.isUndefined(attribute.value) && attribute.value.toString().length > 0 ? (attribute.name + ('="' + attribute.value + '"')) : attribute.name);
```

Comment:
Since we're using "safe" here, we need to make sure these values are actually safe. The attribute name and value should be passed through `apos.utils.escapeHtml`.